vllm/v1/core/scheduler.py

```diff
@@ -420,7 +417,13 @@ def update_from_output(
         # expensive operations inside the loop.
         for request in self.running:
             req_id = request.request_id
-            request.num_computed_tokens += num_scheduled_tokens[req_id]
+            num_tokens_scheduled = num_scheduled_tokens.get(req_id, 0)
+            if num_tokens_scheduled == 0:
+                # The request was not scheduled in this step.
+                new_running.append(request)
+                continue
```

Comment:
Note: Later on when making the scheduler to support PP, I plan to introduce another queue like `self.scheduled` (or other naming) for the actually list of requests being scheduled. The logic here can also be simplified a bit accordingly.